src/io/flutter/preview/PreviewArea.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.preview;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.dartlang.analysis.server.protocol.FlutterOutline;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.HashMap;
+import java.util.Map;
+
+// TODO: have a way to tell the panel whether it's hidden or shown
+
+// TODO: we'll need to know whether the preview element is stateless widget or a state of a stateful
+//       widget; for the 2nd case, we'll need to find the cooresponding stateful widget class
+
+// TODO: we want to preview anything in a state, stateful, or stateless class (not
+//       just things contained in a build method)
+
+// TODO: we should be bolding stateful and stateless (and state) classes, not build() methods
+//       or, show all elements of these classes with some additional emphasis (italic? background color?)
+
+public class PreviewArea {
+  public static int BORDER_WITH = 5;
+
+  @SuppressWarnings("UseJBColor")
```

Comment:
Maybe pastels are OK but in general, how does this do with different themes?  We might consider going to `JBColor` if we want the colors to be theme aware...  Dunno.

Reply:
I don't know any existing code that uses large colorer panels, so there is nothing to copy colors from. I will leave it as is for now, maybe someone will want to experiment with colors for `Darcula`. Or we might get actual rendering (which might be also not dark).